pkg/driver/node/mounter/systemd_mounter.go

```diff
@@ -196,5 +189,27 @@ func (m *SystemdMounter) Unmount(target string) error {
 	if output != "" {
 		klog.V(5).Infof("umount output: %s", output)
 	}
+
+	err = m.credProvider.Cleanup(credentialCtx)
+	if err != nil {
+		klog.V(4).Infof("Unmount: Failed to clean up credentials for %s: %v", target, err)
+	}
+
 	return nil
 }
+
+func (m *SystemdMounter) credentialWriteAndEnvPath() (writePath string, envPath string) {
+	// This is the plugin directory for CSI driver mounted in the container.
+	writePath = "/csi"
+	// This is the plugin directory for CSI driver in the host.
+	envPath = hostPluginDirWithDefault()
+	return writePath, envPath
+}
+
+func hostPluginDirWithDefault() string {
+	hostPluginDir := os.Getenv("HOST_PLUGIN_DIR")
+	if hostPluginDir == "" {
+		hostPluginDir = "/var/lib/kubelet/plugins/s3.csi.aws.com/"
+	}
+	return hostPluginDir
+}
```

Comment:
This two path, `writePath` and `envPath` was hard-coded in `CredentialProvider` before and now an argument to `credentialProvider.{Provide, Cleanup}` as they will be different in `PodMounter`.